### DISCUSSION TITLE
The depth camera doesn't detect weather effects

### DISCUSSION THREAD

**Discussion by Deepakgthomas** [microsoft/AirSim#3695] [Support Q&A]
- AirSim Version/#commit: 1.30
- UE/Unity version: 4.25.4
- autopilot version: NA
- OS Version: Windows 10

### What's the issue you encountered?

While changing the weather, I see that the depth and segmentation cameras do not show the weather change. 
Weather 1 Image - 

![image](https://user-images.githubusercontent.com/21986394/117716383-2397cf00-b19f-11eb-9931-d36005959e5c.png)
It can be argued that the depth camera can't detect rain/snow.

Weather 2 Image (Super Foggy) - 

![image](https://user-images.githubusercontent.com/21986394/117716593-68bc0100-b19f-11eb-99d0-79d072bf44b2.png)

This rightmost camera can detect fog. However, the depth and segmentation camera still behave as if there is no fog.

### How can the issue be reproduced?

Here is my code to get the weather - 

```
import airsim


client = airsim.MultirotorClient()
client.simEnableWeather(True)

client.simSetWeatherParameter(airsim.WeatherParameter.Rain, 100);
client.simSetWeatherParameter(airsim.WeatherParameter.Snow, 300);
client.simSetWeatherParameter(airsim.WeatherParameter.Fog, 50);



```

**Comment by Deepakgthomas** [comment]
@jonyMarino Can you please help with this issue? I hope you don't mind being tagged :D

**Comment by jonyMarino** [comment]
Hi @Deepakgthomas! It may be a non-trivial fix. We will take a look.

**Comment by Deepakgthomas** [comment]
Thanks for replying Jonathan! The depth images are used as input to my RL
algorithm. I wanted to create some kind of noise in my environment such as
snow, rain, leaf etc. Basically change the condition/ weather of my
environment.

**Comment by zimmy87** [comment]
Hi @Deepakgthomas, we experimented with modifying all the collision parameters inside the particle system for snow: ParticleSystem'/AirSim/Weather/WeatherFX/Particles/P_Weather_SnowFX.P_Weather_SnowFX'. Unfortunately, we couldn't figure out a configuration that would make the particles visible to the DepthViz or Segmentation post process materials. We believe that the DepthViz depends on getting the position of the individual particles which is not currently accessible in the WeatherFX. It seems like it would be easier to generate noise for these visualizations in a different way, and you are welcome to submit a feature request.

**Comment by saihv** [comment]
Opened https://github.com/microsoft/AirSim/pull/3699 for a way to get leaves/snow to show up in depth and segmentation.
